src/kernels/scaledkernel.jl

```diff
@@ -0,0 +1,20 @@
+struct ScaledKernel{Tk<:Kernel,Tσ<:Real} <: Kernel
+    kernel::Tk
+    σ::Vector{Tσ}
```

Comment:
I'm curious, why is the scaling parameter a vector? Couldn't you just use
```suggestion
    σ::Tσ
```
?

Reply:
The idea is that it makes the types much more Zygote compatible. One can then optimize the parameters in place, by passing the reference of the parameter. A scalar would not allow this. One other option would be to use a `Ref{Ts}` but I find this quite ugly in general!

Reply:
Hmm, I'm pretty sure that not using a vector should be completely fine with Zygote - at least I would be surprised if it's not: https://fluxml.ai/Zygote.jl/dev/#Structs-and-Types-1

Reply:
Sorry I was not clear. It's actually to be more compatible with `Flux`. 
The whole implicit gradient approach allows you to save the reference to parameters (via an `IdDict` and apply the gradients directly in place without having to build new objects. But this is only possible on mutable object so for a scalar it would not work. Or maybe I got everything wrong about Flux :sweat_smile:

Reply:
What's the issue with using a `Ref` rather than a `Vector` here? I would have thought that a reference to a `Real` would be more readable than a `Vector{<:Real}` that you have to know is  always of length `1`?